22006 beta (#2255)
OpenLiberty/blogs: posts/2022-05-12-stacktrace-log-22006-beta.adoc

----

This new functionality is enabled by configuring either a bootstrap property, an environment variable, or through the `server.xml` file. If configuration is present in all these options, the configuration precedence is the `server.xml` > environment variable > bootstrap property.

*Configuration:*

- `bootstrap.properties`: 
[source]
----
`com.ibm.ws.logging.stackJoin=true
----
- `server.env`: 
[source]
----

> WLP_LOGGING_STACK_JOIN=true/false

Reviewer: why is `true/false` specified in 2/3 of these examples if that's not an actual value and the default is false? I suppose it could be to demonstrate that one config source could override another in some scenarios? But seems a little muddled without more explanation

Author: I agree, I restructured this section with the intent to replace them with just `true` to show how to enable the functionality, but apparently only changed it for the first one.  Going to make that change now

> WLP_LOGGING_STACK_JOIN=true/false

Author: ```suggestion
WLP_LOGGING_STACK_JOIN=true
```

----
- `server.xml`: 
[source,xml]
----

> <logging stackJoin="true/false" />

Author: ```suggestion
<logging stackJoin="true" />
```

---
layout: post

> title: "New Logging functionality to join stack traces in logs in Open Liberty 22.0.0.6-beta"

Reviewer: ```suggestion
title: "Merge stack traces into a single log event in Open Liberty 22.0.0.6-beta"
```

Not wrong as is, but I had to read it twice to understand

# Do NOT change the categories section
categories: blog
author_picture: https://avatars3.githubusercontent.com/ryan-storey
author_github: https://github.com/ryan-storey

> seo-title: New Logging functionality to join stack traces in logs in Open Liberty 22.0.0.6-beta - OpenLiberty.io

Reviewer: ```suggestion
seo-title: Merge stack traces into a single log event in Open Liberty 22.0.0.6-beta - OpenLiberty.io
```

> seo-description: Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs, allowing for the stack trace emitted to be output as a single log event.

Reviewer: ```suggestion
seo-description: Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs so that they are output as a single log event.
```

> blog_description: Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs, allowing for the stack trace emitted to be output as a single log event.

Reviewer: ```suggestion
blog_description: Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs so that they are output as a single log event.
```

open-graph-image: https://openliberty.io/img/twitter_card.jpg
---

> = New Logging functionality to join stack traces in logs in Open Liberty 22.0.0.6-beta

Reviewer: ```suggestion
= Merge stack traces into a single log event in Open Liberty 22.0.0.6-beta
```

Ryan Storey <https://github.com/ryan-storey>
:imagesdir: /
:url-prefix:
:url-about: /
//Blank line here is necessary before starting the body of the post.

> Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs, allowing for the stack trace emitted to be output as a single log event.

Reviewer: ```suggestion
Open Liberty 22.0.0.6-beta offers one new beta feature (along with all GA features), which provides new logging functionality to join stack traces in logs so that they are output as a single log event.
```

> The link:{url-about}[Open Liberty] 22.0.0.6-beta includes the following beta features (along with all GA features):

Reviewer: ```suggestion
The link:{url-about}[Open Liberty] 22.0.0.6-beta includes the following beta feature (along with all GA features):
```

* <<logging, New Logging functionality to join stack traces in logs>>

[#logging]

> == New Logging functionality to join stack traces in logs

Reviewer: ```suggestion
== New Logging functionality: merge stack traces into a single log event
```

> Prior to this feature, when a stack trace was logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, each line of the stack trace was printed as a separate event.

Reviewer: ```suggestion
```

> The newly introduced functionality allows for the emitted stack trace to be output as a single log event. This is most impactful to users who forward their logs down stream to third party log analysis technologies. An example of this being the (E)lastic (L)ogstash (K)ibana stack.   This affects the console output, console logs , messages log and the trace logs.

Reviewer: ```suggestion
When a stack trace is logged in Open Liberty, typically from `Throwable.printStackTrace()` calls from the application, you can now output the emitted stack trace as a single log event. This update is impactful if you forward your logs downstream to third-party log analysis technologies, such as the Elastic Logstash Kibana (ELK) stack. It affects the console output, console logs , messages log and the trace logs. Prior to this update, each line of the stack trace was printed as a separate event.
```

----

> This new functionality is enabled by configuration through either the a bootstrap property, environment variable and/or through the `server.xml`. Note that if there is configuration present in all strategies mentioned then the configuration precedence is the `server.xml` > environment variable > bootstrap property.

Reviewer: ```suggestion
This new functionality is enabled by configuring either a bootstrap property, an environment variable, or through the `server.xml` file. If configuration is present in all these options, the configuration precedence is the `server.xml` > environment variable > bootstrap property.
```
